### PR TITLE
BaSP-TG-03: Delete and update employees

### PR DESCRIPTION
**Pull request by ZGMauri**
Methods to delete and update employees

### CODE REVIEW

File: src/controllers/employees.js [hunk 1]
```diff
@@ -1,156 +1,48 @@
...
-    res.send(filteredRole);
-  } else {
-    res.send('Role not found');
+    return res.status(200).json({
+      message: `Employee with id ${id} deleted`,
+      data: result,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(400).json({
```
Comment (martinraveglia) [inline src/controllers/employees.js:14]: If no employee with the id given was found, the status code should be 404

File: src/controllers/employees.js [hunk 2]
```diff
@@ -1,156 +1,48 @@
...
-        res.send('Employee deleted.');
-      }
+};
+
+const updateEmployees = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const result = await Employees.findByIdAndUpdate(
+      { _id: id },
+      { ...req.body },
```
Comment (martinraveglia) [inline src/controllers/employees.js:27]: why are you destructuring this like that? It's the same as passing just `req.body`

File: src/controllers/employees.js [hunk 3]
```diff
@@ -1,156 +1,48 @@
...
+      { new: true },
+    );
+
+    return res.status(200).json({
+      message: `Employee with id ${id} edited`,
+      date: result,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(400).json({
```
Comment (martinraveglia) [inline src/controllers/employees.js:37]: If no employee with the id given was found, the status code should be 404

File: src/controllers/timeSheets.js [hunk 4]
```diff
@@ -1,87 +0,0 @@
-import express from 'express';
```
Comment (martinraveglia) [inline src/controllers/timeSheets.js:1]: Why did you erase this file content? It should remain untouched as your feature is not related to timeSheets

File: src/validations/employees.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  });
+
+  const validation = employeesValidations.validate(req.body);
+
+  if (validation.error) {
+    return res.status(400).json({
+      message: `There was an error: ${validation.error.details[0].message}`,
+      data: undefined,
+      error: true,
+    });
```
Comment (martinraveglia) [inline src/validations/employees.js:19]: if we have more than one validation error (in more than one field), all errors should be sent

File: src/validations/employees.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  }
+  return next();
+};
+
+export default {
+  validateUpdate,
+};
```
Comment (martinraveglia) [inline src/validations/employees.js:26]: We are exporting only one method, exporting it as an object is not necessary

**Review by martinraveglia — CHANGES_REQUESTED** [review]

File: src/controllers/employees.js [hunk 7]
```diff
@@ -1,156 +1,60 @@
...
-    } else {
-      res.send('Employee updated successfully');
+    return res.status(200).json({
+      message: `Employee with id ${id} deleted`,
+      data: result,
+      error: false,
+    });
+  } catch (error) {
+    let status = 404;
+    if (error.message.includes('Employee not found')) {
```
Comment (martinraveglia) [inline src/controllers/employees.js:17]: this will never match the error

File: src/controllers/employees.js [hunk 8]
```diff
@@ -1,156 +1,60 @@
...
-    if (e.projects.find((r) => r.role === roleParam)) {
-      filteredRole.push(e);
+    return res.status(200).json({
+      message: `Employee with id ${id} edited`,
+      date: result,
+      error: false,
+    });
+  } catch (error) {
+    let status = 404;
+    if (error.message.includes('Employee not found')) {
```
Comment (martinraveglia) [inline src/controllers/employees.js:46]: This will never match the error

File: src/routes/employees.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+import { deleteEmployees, updateEmployees } from '../controllers/employees';
+import employeesValidations from '../validations/employees';
+
+const router = express.Router();
+
+router.delete('/:id', deleteEmployees);
+router.put('/:id', employeesValidations.validateUpdate, updateEmployees);
```
Comment (martinraveglia) [inline src/routes/employees.js:8]: this is wrongly used now

**Review by martinraveglia — CHANGES_REQUESTED** [review]

File: src/routes/employees.js (hunk 9 above)
Comment (ZGMauri) [reply src/routes/employees.js:8]: yes I saw this error while correcting the other PR

File: src/controllers/employees.js [hunk 10]
```diff
@@ -1,156 +1,60 @@
...
-      res.send('Employee updated successfully');
+    return res.status(200).json({
+      message: `Employee with id ${id} deleted`,
+      data: result,
+      error: false,
+    });
+  } catch (error) {
+    let status = 404;
+    if (!error.message.includes('Employee not found')) {
+      status = 404;
```
Comment (martinraveglia) [inline src/controllers/employees.js:18]: Status is always 404 with this logic, also, the error message will never match. Please try to understand how this logic works!

```suggestion
    let status = 400;
    if (error.message.includes('Employee with id')) {
      status = 404;
```

File: src/controllers/employees.js [hunk 11]
```diff
@@ -1,156 +1,60 @@
...
-    if (e.projects.find((r) => r.role === roleParam)) {
-      filteredRole.push(e);
+    return res.status(200).json({
+      message: `Employee with id ${id} edited`,
+      date: result,
+      error: false,
+    });
+  } catch (error) {
+    let status = 400;
+    if (!error.message.includes('Employee not found')) {
```
Comment (martinraveglia) [inline src/controllers/employees.js:46]: The error message will never match. Please try to understand how this logic works!

```suggestion
    if (!error.message.includes(`Employee ${req.query.id} not found`)) {
```

**Review by martinraveglia — CHANGES_REQUESTED** [review]

**Review by martinraveglia — APPROVED** [review]

**Pull request merged**